lesson-7/script.js

```diff
@@ -0,0 +1,157 @@
+const isNumber = (number) => {
+    return typeof (number) === 'number' && !isNaN(number)
+}
+const isString = (string) => {
+    return typeof (string) === 'string' && !!string
+}
+const isBoolean = (boolean) => {
+    return typeof (boolean) === 'boolean'
+}
+/* 
+1. Cтворити функцію, яка створює та реєструє нову команду у автопробізі. 
+Під час виклику функції потрібно передати дані про команду - назва команди, ім'я водія, 
+рік народження водія, марка автомобіля, спонсор команди та чи водій допущений до автопробігу
+*/
+const createNewTeam = (team_name, driver_name, driver_birth_year, car_mark, team_sponsor, driver_admission) => {
+    if (isString(team_name) || !isString(driver_name) || !isNumber(driver_birth_year) || !isString(car_mark) || !isString(team_sponsor) || !isBoolean(driver_admission)) {
+        return 'Some arguments has an incorrect type.'
+    } else if (driver_birth_year >= 2004 && driver_birth_year <= 1962) {
+        return 'It seems that driver year of birth is not real or allowed.'
+    }
+    return {
+        team_name,
+        driver_name,
+        driver_birth_year,
+        car_mark,
+        team_sponsor,
+        driver_admission,
+    }
+}
+const team_1 = createNewTeam ('1', 'ros', 1994, 'true', 'ford', true);
+/* 
+2. Запитати дані (через prompt та циклічно) про нового користувача у системі та зберегти його в об'єкт. 
+Дані про користувача - логін, пароль, місто, країна, стать, вік
+*/
+let newUserTemplate = {
+    login: "empty",
+    password: "empty",
+    city: "empty",
+    country: "empty",
+    gender: "empty",
+    age: "empty",
+}
+const createUser = () => {
+    let newUser = {}
+    for (let key in newUserTemplate) {
+        let collectUserData = prompt(`Enter your ${key}, please.`);
+        if (key === 'login' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'password' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'city' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'country' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'gender' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'age' && !isNumber(Number(collectUserData))) {
+            return console.log('Some arguments has an incorrect type.');
```

Comment:
This will give a result value as `undefined`. No need to return anything in that case

```suggestion
            console.log('Some arguments has an incorrect type.');
            return;
```